crypto/ec/ec_kmeth.c

```diff
@@ -139,8 +140,10 @@ int ECDH_compute_key(void *out, size_t outlen, const EC_POINT *pub_key,
         ECerr(EC_F_ECDH_COMPUTE_KEY, EC_R_INVALID_OUTPUT_LENGTH);
         return 0;
     }
-    if (!eckey->meth->compute_key(&sec, &seclen, pub_key, eckey))
-        return 0;
+    compute_key_return = 
+        eckey->meth->compute_key(&sec, &seclen, pub_key, eckey);
+    if (compute_key_return <= 0)
+        return compute_key_return;
```

Comment:
I suggest you change the check from `!` to `< 0` and then just return -1, no need for the variable.

Reply:
Good point. I will change the return of the other error cases to "-1" too.